Fitness/Controllers/ProBioViewController.swift

```diff
@@ -0,0 +1,268 @@
+        return CGSize(width: collectionView.frame.width, height: 435)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        let bioTextViewHorizontalPadding = 40
+        let bioTextViewTopPadding = 32
+        let bioSummaryHorizontalPadding = 40
+        let bioSummaryTopPadding = 64
+        let dividerViewHeight = 1
+        let dividerViewTopPaddingLarge = 36
+
+        let dividerViewTopPaddingSmall = 23
+        let descriptionLabelHorizontalPadding = 48
+        let descriptionLabelTopPadding = 16
+        let expertiseLabelHeight = 20
+        let expertiseLabelTopPadding = 23
+
+        let routinesCollectionViewBottomPadding = 61
+        let routinesLabelHeight = 20
+        let routinesLabelTopPadding = 24
+
+        let linksCollectionViewBottomPadding = 57
+        let linksCollectionViewCellSize = 24
+        let linksCollectionViewTopPadding = 12
+        let linksTitleLabelHeight = 18
+
+        let width = collectionView.frame.width
+        let itemType = sections[indexPath.section].items[indexPath.item]
+        switch itemType {
+        case .biography:
+            let bioSummaryWidth = width - CGFloat(bioSummaryHorizontalPadding * 2)
+            let bioSummaryHeight = pro.summary.height(withConstrainedWidth: bioSummaryWidth, font: UIFont._20MontserratSemiBold!)
+            let bioTextViewWidth = width - CGFloat(bioTextViewHorizontalPadding * 2)
+            let bioTextViewHeight = pro.bio.height(withConstrainedWidth: bioTextViewWidth, font: UIFont._14MontserratLight!)
+            let textHeight = CGFloat(bioSummaryTopPadding + bioTextViewTopPadding) + bioSummaryHeight + bioTextViewHeight
+            let height = textHeight + CGFloat(dividerViewTopPaddingLarge + dividerViewHeight)
+            return CGSize(width: width, height: height)
+        case .expertise:
+            let descriptionLabelWidth = width - CGFloat(descriptionLabelHorizontalPadding * 2)
+            let descriptionLabelHeight = pro.expertise.joined(separator: " · ").height(withConstrainedWidth: descriptionLabelWidth, font: UIFont._14MontserratLight!)
+            let height = CGFloat(expertiseLabelTopPadding + expertiseLabelHeight + descriptionLabelTopPadding + dividerViewTopPaddingSmall + dividerViewHeight) + descriptionLabelHeight
+            return CGSize(width: width, height: height)
+        case .routines:
+            let height = routinesCollectionViewHeight() + CGFloat(routinesLabelTopPadding + routinesLabelHeight + routinesCollectionViewBottomPadding)
+            return CGSize(width: width, height: height)
+        case .links:
+            let height = CGFloat(linksTitleLabelHeight + linksCollectionViewTopPadding + linksCollectionViewCellSize + linksCollectionViewBottomPadding)
+            return CGSize(width: width, height: height)
```

Comment:
Would it be a bad idea to move this logic into the actual cell files and store the values as a constant like `ProBioLinksCell.height`? Cuz it seems like each of these heights are static and they correspond to the collectionview cell being used at that indexPath @kevinchan159

Reply:
i like that +1

Reply:
what would be the best place within the cell files to do the calculation?